v2/tests/log_context_property_basic_types_ut/log_context_property_basic_types_ut.c

```diff
@@ -309,37 +309,1626 @@ static void int32_t_free_returns(void)
-    int32_t_copy_called_with_NULL_src_value_fails();
-    int32_t_copy_succeeds();
-    int32_t_copy_succeeds_2();
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string(NULL, buffer, TEST_BUFFER_SIZE);
 
-    int32_t_free_returns();
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_012: [ LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t_t).to_string shall call snprintf with buffer, buffer_length and format string PRIu64 and pass in the values list the uint64_t_t value pointed to be property_value. ]*/
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_013: [ LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t_t).to_string shall succeed and return the result of snprintf. ]*/
+static void uint64_t_to_string_succeeds(void)
+{
+    // arrange
+    char buffer[TEST_BUFFER_SIZE];
+    uint64_t uint64_t_value = 0;
+
+    setup_mocks();
+    setup_expected_snprintf_call();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string(&uint64_t_value, buffer, TEST_BUFFER_SIZE);
+
+    // assert
+    POOR_MANS_ASSERT(result == 1);
+    POOR_MANS_ASSERT(strcmp(buffer, "0") == 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_012: [ LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t_t).to_string shall call snprintf with buffer, buffer_length and format string PRIu64 and pass in the values list the uint64_t_t value pointed to be property_value. ]*/
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_013: [ LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t_t).to_string shall succeed and return the result of snprintf. ]*/
+static void uint64_t_to_string_UINT64_MAX_succeeds(void)
+{
+    // arrange
+    char buffer[TEST_BUFFER_SIZE];
+    uint64_t uint64_t_value = UINT64_MAX;
+
+    setup_mocks();
+    setup_expected_snprintf_call();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string(&uint64_t_value, buffer, TEST_BUFFER_SIZE);
+
+    // assert
+    POOR_MANS_ASSERT(result == 20);
+    POOR_MANS_ASSERT(strcmp(buffer, "18446744073709551615") == 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_012: [ LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t_t).to_string shall call snprintf with buffer, buffer_length and format string PRIu64 and pass in the values list the uint64_t_t value pointed to be property_value. ]*/
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_013: [ LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t_t).to_string shall succeed and return the result of snprintf. ]*/
+static void uint64_t_to_string_with_truncation_succeeds(void)
+{
+    // arrange
+    char buffer[2];
+    uint64_t uint64_t_value = 42;
+
+    setup_mocks();
+    setup_expected_snprintf_call();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string(&uint64_t_value, buffer, 2);
```

Comment:
>2 [](http://example.com/codeflow?start=93&length=1)

this could be expressed as sizeof(buffer).

Consider here and in all the other tests.

Reply:
Changed to sizeof(buffer) all over the place.